sysboundary/sysboundarycondition.cpp

```diff
@@ -785,48 +785,60 @@ namespace SBC {
       return flowtoCellsBlock;
    }
    
-   Real SysBoundaryCondition::fieldBoundaryCopyFromExistingFaceNbrMagneticField(
+   Real SysBoundaryCondition::fieldBoundaryCopyFromSolvingNbrMagneticField(
       FsGrid< std::array<Real, fsgrids::bfield::N_BFIELD>, 2> & perBGrid,
       FsGrid< fsgrids::technical, 2> & technicalGrid,
       cint i,
       cint j,
       cint k,
-      cuint component
+      cuint component,
+      cuint mask
    ) {
-      const std::array<int,3> closestCell = getTheClosestNonsysboundaryCell(technicalGrid, i, j, k);
-      
-      const std::array<int32_t, 3> gid = technicalGrid.getGlobalIndices(i, j, k);
-      const std::array<int32_t, 3> ngid = technicalGrid.getGlobalIndices(closestCell[0], closestCell[1], closestCell[2]);
 
-      if (closestCell[0] == std::numeric_limits<int>::min()) {
-         //cerr << "(" << gid[0] << "," << gid[1] << "," << gid[2] << ")" << __FILE__ << ":" << __LINE__ << ": No closest cell found!" << endl;
-         //abort();
+      int distance = std::numeric_limits<int>::max();
+      std::vector< std::array<int,3> > closestCells;
 
+      for (int kk=-2; kk<3; kk++) {
+         for (int jj=-2; jj<3; jj++) {
+            for (int ii=-2; ii<3 ; ii++) {
+               if( technicalGrid.get(i+ii,j+jj,k+kk) // skip invalid cells returning NULL
+                   && (technicalGrid.get(i+ii,j+jj,k+kk)->SOLVE & mask) == mask // Did that guy solve this component?
+                   && technicalGrid.get(i+ii,j+jj,k+kk)->sysBoundaryFlag != sysboundarytype::DO_NOT_COMPUTE // Do not copy from there
+                   && ii != 0 && jj != 0 && kk != 0 // skip oneself!
```

Comment:
Of this logic, I added also the last line but it is unnecessary as a cell calling this function wouldn't match the mask condition. But something (here or elsewhere) is flawed as I don't get the correct copied values e.g. for fg_per_b_x at the -z edge.

Reply:
So what was the thing here? Make clean? :)

Reply:
ping @ykempf

Reply:
No, removing the last line of checks as that would trigger whenever ii, jj or kk was zero, so copying from any but face neighbours.

Reply:
ahh right :)